testscripts/stor_reset_scsi.ps1

```diff
@@ -0,0 +1,258 @@
+    }
+    else {
+        "Warn : rootdir '${rootDir}' does not exist"
+    }
+}
+
+
+# Source the tcutils.ps1 file
+. .\setupscripts\tcutils.ps1
+
+PowerCLIImport
+ConnectToVIServer $env:ENVVISIPADDR `
+    $env:ENVVISUSERNAME `
+    $env:ENVVISPASSWORD `
+    $env:ENVVISPROTOCOL
+
+
+###############################################################################
+# Main Body
+###############################################################################
+$retVal = $Failed
+$vmObj = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+if (-not $vmObj) {
+    Write-Host -F Red "ERROR: Unable to Get-VM with $vmName"
+    Write-Output "ERROR: Unable to Get-VM with $vmName"
+    DisconnectWithVIServer
+    return $Aborted
+}
+
+
+# Get the Guest version
+$DISTRO = GetLinuxDistro ${ipv4} ${sshKey}
+Write-Host -F Red "DEBUG: DISTRO: $DISTRO"
+Write-Output "DEBUG: DISTRO: $DISTRO"
+if (-not $DISTRO) {
+    Write-Host -F Red "ERROR: Guest OS version is NULL"
+    Write-Output "ERROR: Guest OS version is NULL"
+    DisconnectWithVIServer
+    return $Aborted
+}
+Write-Host -F Red "INFO: Guest OS version is $DISTRO"
+Write-Output "INFO: Guest OS version is $DISTRO"
+
+
+# Different Guest DISTRO, different modules
+if ($DISTRO -ne "RedHat7" -and $DISTRO -ne "RedHat8" -and $DISTRO -ne "RedHat6") {
+    Write-Host -F Red "ERROR: Guest OS ($DISTRO) isn't supported, MUST UPDATE in Framework / XML / Script"
+    Write-Output "ERROR: Guest OS ($DISTRO) isn't supported, MUST UPDATE in Framework / XML / Script"
+    DisconnectWithVIServer
+    return $Skipped
+}
+
+
+# Poweroff the VM
+Stop-Vm $vmobj -Confirm:$false -RunAsync:$true -ErrorAction SilentlyContinue
+# Fresh the vm status
+$vmObj = Get-VMHost -Name $hvServer | Get-VM -Name $vmName
+
+
+# add multiple disks 2 times
+for ($i = 1; $i -le 2; $i++) 
+{
+    # Add Disk with size 5 to 10
+    $hd_size = Get-Random -Minimum 5 -Maximum 10
+    Write-Host -F Red "INFO: Add $hd_nim disk(s) with size $hd_size GB to the VM $vmName"
+    Write-Output "INFO: Add $hd_nim disk(s) with size $hd_size GB to the VM $vmName"
+    New-HardDisk -CapacityGB $hd_size -VM $vmObj -StorageFormat "Thin" -Controller "SCSI Controller 0" -ErrorAction SilentlyContinue
+    if (-not $?) 
+    {
+        Write-Host -F Red "ERROR: Cannot add new hard disk to the VM $vmName"
+        Write-Output "ERROR: Cannot add new hard disk to the VM $vmName"
+        DisconnectWithVIServer
+        return $retVal
+    }
+    Start-Sleep -Seconds 1
+}
+# PowerOn the VM
+Write-Host -F Green "DEBUG: start the VM $vmName"
+Write-Output "DEBUG: start the VM $vmName"
+Start-Vm $vmobj -Confirm:$false -RunAsync:$true -ErrorAction SilentlyContinue
+# Wait for complete start vm
+$ret = WaitForVMSSHReady $vmName $hvServer ${sshKey} 300
+if ( $ret -ne $true )
+{
+    Write-Output "Failed: Failed to start VM."
+    write-host -F Red "Failed: Failed to start VM."
+    DisconnectWithVIServer
+    return $Aborted
+}
+
+
+$disk = ("sdb","sdc")
```

Comment:
After snapshot, VM only has sda, so sdb, sdc can be used by correctly.
But i hope you can get these disk name dynamically.